### PR TITLE
Fix server logging; lower threads from 16 -> 3

### PR DESCRIPTION
**Pull request by mark-idleman**
- Fixes logging config so we're not getting spammed by DEBUG level netty logs (thousands/second)
- Changes server threads from 16->3, as discussed [here](https://replicahq.slack.com/archives/C01JSFB04B1/p1613075123053600?thread_ts=1613073872.052700&cid=C01JSFB04B1)
- Increases connection max grace period (time that server allows current RPC connection to keep running after server hits max connection age of 10s and tries to cut connection); seems like upping this might help our RPC error situation/give us more breathing room
- Adds actual logging for all cases where requests aren't successful (these should be only logs that show up now, aside from basic INFO logging on router startup)

New logs look like this:
Screen Shot 2021-02-11 at 3 06 36 PM

### CODE REVIEW

File: grpc/src/main/java/RouterServer.java [hunk 1]
```diff
@@ -107,8 +107,8 @@ private void start() throws IOException {
                 .addService(new RouterImpl(graphHopper, ptRouter, matrixAPI, gtfsLinkMappings, gtfsRouteInfo, gtfsFeedIdMapping))
                 .addService(ProtoReflectionService.newInstance())
                 .maxConnectionAge(10, TimeUnit.SECONDS)
-                .maxConnectionAgeGrace(10, TimeUnit.SECONDS)
-                .executor(Executors.newFixedThreadPool(16))
+                .maxConnectionAgeGrace(30, TimeUnit.SECONDS)
+                .executor(Executors.newFixedThreadPool(3))
```
Comment (JacobHayes) [inline grpc/src/main/java/RouterServer.java:111]: Was there some other config (maybe xml?) where we could tune threads like listener vs worker vs etc? If so, how do these interact?

**Review by JacobHayes — APPROVED** [review]
Sweet, lgtm! I think we may get the new log messages twice (once on the router and once on the worker), but maybe that's ok. I wonder if metrics tracking these would be useful (ie: help us notice if the network is bad or unroutable for some upstream reason?)

File: grpc/src/main/java/RouterServer.java (hunk 1 above)
Comment (mark-idleman) [reply grpc/src/main/java/RouterServer.java:111]: For the old http version, you would set Jetty server settings in the main graphhopper config file (yaml), but for the new grpc version, all server-level settings are set here in the code. So basically, what's in these few lines is all that's used to tune the server. There's loads more that can be set: https://grpc.github.io/grpc-java/javadoc/io/grpc/netty/NettyServerBuilder.html

**Comment by mark-idleman** [comment]
> I think we may get the new log messages twice

Yes, kinda - the server will print more detailed info (like full stack traces when real java errors are thrown, etc), while the client has its own [code](https://github.com/replicahq/model/blob/master/mapservice/api/graphhopper_client.py#L113-L126) for catching the server errors + logging (it's a lot more high-level, usually single-line simple log messages based on the case). I'd argue that having both is standard practice and how we should have things set up; our debugging flow should go (see client had an error) -> (check service logs for root cause, if unclear, based on error and how client handles it)

> I wonder if metrics tracking these would be useful

What do you envision here, a counter of each type of log thrown, both client-side and server-side? I can see how that could help catching inconsistencies, but it feels like the logic is simple enough now that we can account for all cases easily just looking at the client/server error handling code (there's only 2 or 3 possible situations, which should be accounted for on both ends now)

**Comment by JacobHayes** [comment]
> the server will print more detailed info [...] having both is standard practice

Sweet, sgtm! I'm probably slightly of the "either log it or handle it, but not both" logging philosophy, but the added context in networked contexts seems helpful, anything to make the actionable issues more visible! :)

> What do you envision here

Yeah a counter, but probably don't need it on both ends - I was just curious to know how many times we get these unroutable requests in general (maybe I'm in the "instrument all the things" phase now haha). I guess it'd spike with larger regions/builds so maybe it's not particularly actionable (unless we could get "unroutable / total" error rate), but it maybe it'd help with figuring out whether routable_cells is important or not for example (if that's still around and there aren't better ways to tell).

**Comment by mark-idleman** [comment]
Totally makes sense re: your logging philosophy haha (I shouldn't claim there's one "standard practice" for such a big topic 😄). We're just coming from a world where we were accidentally swallowing a lot of errors/not logging them when they really needed to be logged (NPEs, for example...), along with all the cluttered debug logs, so I think this is a happy medium at the moment.

And gotcha, totally makes sense! You're actually in [luck](https://app.datadoghq.com/dashboard/vtr-jnt-j56/replicabuilder-metrics?from_ts=1612989749157&fullscreen_section=overview&fullscreen_widget=4238947753689584&live=true&to_ts=1613076149157&tpl_var_deployment=rockwellweiner-078bb181-routerservicegh-socal&tpl_var_replica_region=socal&tv_mode=false&fullscreen_start_ts=1613011383240&fullscreen_end_ts=1613097783240&fullscreen_paused=false)! The exact logic for the lables can be figured out by looking at the error handling in `RouterImpl.java` (server) and `graphhopper_client.py` (client), but the basic gist is that `routes_found` being `true` means success, `false` means "not found" (so, couldn't snap o/d to map, or just couldn't find suitable route), and "error" (any other type of error, so, all RPC errors, java exceptions in the server, etc). These correspond to `_status` [here](https://github.com/replicahq/model/blob/master/mapservice/api/graphhopper_client.py#L129-L133)

**Comment by JacobHayes** [comment]
Oooh wow that's great! 😁

**Review by kaelgreco — APPROVED** [review]
lgtm!
let's communicate that when we land the change w/ the new image we can just copy sources.

**Review by hardisty — APPROVED** [review]
LGTM

File: grpc/src/main/java/RouterImpl.java [hunk 2]
```diff
@@ -128,7 +128,7 @@ public void routeStreetMode(StreetRouteRequest request, StreamObserver<StreetRou
             String message = "GH internal error! Path could not be found between "
                     + ghRequest.getPoints().get(0).lat + "," + ghRequest.getPoints().get(0).lon + " to "
                     + ghRequest.getPoints().get(1).lat + "," + ghRequest.getPoints().get(1).lon;
-            logger.error(message);
+            logger.error(message, e);
```
Comment (JacobHayes) [inline grpc/src/main/java/RouterImpl.java:131]: Does this show the stacktrace? 👍

**Review by JacobHayes — APPROVED** [review]

File: grpc/src/main/java/RouterImpl.java (hunk 2 above)
Comment (mark-idleman) [reply grpc/src/main/java/RouterImpl.java:131]: yup!

Comment (mark-idleman) [reply grpc/src/main/java/RouterImpl.java:131]: We already do it for PT request errors, just forgot to add it for street requests (auto/bike/ped)

**Comment by mark-idleman** [comment]
For context, synced with Jacob and Frank earlier today and we decided that the "route not found" warning logs are worth removing for the time being, as they're 1) still fairly frequent (so possibly slow) 2) don't add much info for us (plus we log these client-side anyway). Real java errors are still caught + logged properly (including full stack trace).

Merging because we seem happy with these tweaks based on recent socal/norcal runs.

**Pull request merged**